modules/geochemistry/include/utils/GeochemistryActivityCoefficientsDebyeHuckel.h

```diff
@@ -0,0 +1,112 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "GeochemistryActivityCoefficients.h"
+#include "GeochemistryIonicStrength.h"
+
+struct DebyeHuckelParameters
+{
+  Real A;
+  Real B;
+  Real Bdot;
+  Real a_water;
+  Real b_water;
+  Real c_water;
+  Real d_water;
+  Real a_neutral;
+  Real b_neutral;
+  Real c_neutral;
+  Real d_neutral;
+
+  DebyeHuckelParameters()
+    : A(0.5092),
+      B(0.3283),
+      Bdot(0.035),
+      a_water(1.45397),
+      b_water(0.022357),
+      c_water(0.0093804),
+      d_water(-0.0005262),
+      a_neutral(0.1127),
+      b_neutral(-0.01049),
+      c_neutral(0.001545),
+      d_neutral(0.0)
+  {
+  }
+};
+
+/**
+ * Computes activity coefficients for non-minerals and non-gases (since these species do not have
+ * activity coefficients). Also computes the activity of water.  Uses a Debye-Huckel model
+ */
+class GeochemistryActivityCoefficientsDebyeHuckel : public GeochemistryActivityCoefficients
+{
+public:
+  /// Method used by this class to compute activity coefficients and activity of water
+  enum class ActivityCoefficientMethodEnum
```

Comment:
I believe this should be in the base class as a virtual. The base class will then (or some helper) hold an enum of all the valid types, and the children classes will override with their type of these.

Reply:
Great, thanks @loganharbour .  I got up especially early to address your change requests, and hopefully we can get this merged today.  I'll work on this request shortly....

Reply:
👍 I'll be around for a while

Reply:
OK, i cannot forward-declare an enum and define it in a derived class.  So I think this suggestion is not going to work.  Also, I don't want derived classes such as `GeochemistryActivityCoefficientsDebyeHuckel` knowing about other activity-coefficient methods that will be used in other derived classes (i know there are no other derived classes at the moment).

Instead, I'm simply going to remove this enum entirely.  It was premature to put it in - i thought that someone might add to it one day to implement different DebyeHuckel-style activity calculators inside `GeochemistryActivityCoefficientsDebyeHuckel`   But i'll let the do that in the future.

Reply:
Okay, that's fine. The way this is typically done is that _some_ object needs to know about all of the possible choices. Typically, we'll declare that in the parent object or using some registry system.

Reply:
OK, cool.  Let's watch those green ticks appear....